src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/ServerModePerCallOutputDevice.cs

```diff
@@ -19,11 +34,72 @@ internal class ServerModePerCallOutputDevice : IPlatformOutputDevice
 
     public Task DisplayAfterSessionEndRunAsync() => Task.CompletedTask;
 
-    public Task DisplayAsync(IOutputDeviceDataProducer producer, IOutputDeviceData data) => Task.CompletedTask;
+    public async Task DisplayAsync(IOutputDeviceDataProducer producer, IOutputDeviceData data)
+    {
+        using (await _asyncMonitor.LockAsync(TimeoutHelper.DefaultHangTimeSpanTimeout))
+        {
+            switch (data)
+            {
+                case FormattedTextOutputDeviceData formattedTextOutputDeviceData:
+                    LogLevel logLevel = formattedTextOutputDeviceData.ForegroundColor is SystemConsoleColor color
+                        ? color.ConsoleColor switch
+                        {
+                            ConsoleColor.Red => LogLevel.Error,
+                            ConsoleColor.Yellow => LogLevel.Warning,
+                            _ => LogLevel.Information,
+                        }
+                        : LogLevel.Information;
+
+                    await LogAsync(logLevel, formattedTextOutputDeviceData.Text, formattedTextOutputDeviceData.Padding);
+                    break;
+
+                case TextOutputDeviceData textOutputDeviceData:
+                    await LogAsync(LogLevel.Information, textOutputDeviceData.Text, padding: null);
+                    break;
+
+                case ExceptionOutputDeviceData exceptionOutputDeviceData:
+                    await LogAsync(LogLevel.Error, exceptionOutputDeviceData.Exception.ToString(), padding: null);
+                    break;
+            }
+        }
+    }
 
     public Task DisplayBannerAsync(string? bannerMessage) => Task.CompletedTask;
```

Comment:
Let's handle this case and print as debug.

@MarcoRossignoli @drognanar it feels like it would be too verbose to print as Info for TE but maybe we still prefer doing it. WDYT?